### DISCUSSION TITLE
No sounds comes

### DISCUSSION THREAD

**Discussion by tizfra** [GeekFunkLabs/fluidpatcher#56] [Q&A] [answered]
Hi, I've update my squishbox to version 0.7.12, but now no sounds comes out.
Midi messages arrives correctly to squishbox and also isn't a hardware sound module problem, because i've tryed to reinstall fluidpatcher to my squishbox using raspberry headphone jack and no sounds comes out.
@albedozero if you need more information, tell me what you need.

**Comment by albedozero** [comment]
Hmm.. I haven't found a bug yet. Can you send the _squishboxconf.yaml_ and bank file that you are using?

**Comment by tizfra** [comment]
Yes of course
[squishbox.zip](https://github.com/albedozero/fluidpatcher/files/9314591/squishbox.zip)

**Reply by albedozero** [reply]
You can probably remove the line `synth.audio-channels: 16` from your _squishboxconf.yaml_. It was in there for effects routing, but newer versions don't need it.

Also, I noticed you have a high `synth.gain` value set in your _bank8.yaml_ - values above 1.0 can produce some unwanted distortion. If you're trying to get louder audio output you can try turning up the volume using `alsamixer` at the command line.

**Reply by tizfra** [reply]
How can I turning up volume in `alsamixer`? I've tryed but on sound card `snd_rpi_hifiberry_dac` tells me that this sound decice does not have any controls.

**Comment by albedozero** [comment, marked as answer]
I may have figured it out - Recently after a lot of testing I decided to switch from using the JACK driver (`audio.driver: jack`) and use the default ALSA driver. JACK is meant to provide fast communication when you have multiple audio applications at once, which is not the case in the SquishBox, plus it adds an unnecessary extra layer of processing overhead. However, the installer wasn't automatically making this switch for users who installed using an older version that uses JACK.

To fix, run the installer (which I just updated), and for the question "Which audio output would you like to use?" choose _sndrpihifiberry_ if you want to use the SquishBox's 1/4" outputs, or _Headphones_ if you want to use the physical headphone jack on the Raspberry Pi. You can answer no/no change for the other questions.

You could instead do this manually by changing the line in your _squishboxconf.yaml_
```yaml
  audio.driver: jack
```
to
```yaml
  audio.driver: alsa
```

**Comment by j-vanwijk** [comment]
I have also some problems with the new way. 
I have a compleet new installation.
With a usb audio (Dragonfly) stick.
i have checked with the alsamix to volume and is it's umuted.

Pre versions did work out of the box.

raspmidipi:~ $ aplay -l
**** List of PLAYBACK Hardware Devices ****
card 0: Headphones [bcm2835 Headphones], device 0: bcm2835 Headphones [bcm2835 Headphones]
  Subdevices: 8/8
  Subdevice #0: subdevice #0
  Subdevice #1: subdevice #1
  Subdevice #2: subdevice #2
  Subdevice #3: subdevice #3
  Subdevice #4: subdevice #4
  Subdevice #5: subdevice #5
  Subdevice #6: subdevice #6
  Subdevice #7: subdevice #7
card 2: DragonFly [AudioQuest DragonFly], device 0: USB Audio [USB Audio]
  Subdevices: 1/1
  Subdevice #0: subdevice #0
card 3: vc4hdmi [vc4-hdmi], device 0: MAI PCM i2s-hifi-0 [MAI PCM i2s-hifi-0]
  Subdevices: 1/1
  Subdevice #0: subdevice #0


# config file used by fluidpatcherconf.yaml
soundfontdir: SquishBox/sf2
bankdir: SquishBox/banks
mfilesdir: SquishBox/midi
plugindir: /usr/lib/ladspa
currentbank: bank1.yaml


###### fluidpatcherconf.yaml 
fluidsettings:
  audio.alsa.device: hw:DragonFly
  audio.driver: alsa
  midi.autoconnect: 1
  player.reset-synth: 0
  synth.ladspa.active: 1

**Comment by albedozero** [comment]
I assume you are getting no sound when you play on your connected MIDI keyboard? It could be you have modified the wrong config file - _fluidpatcherconf.yaml_ is the config file for the desktop program _fluidpatcher.pyw_. If you are running _squishbox.py_ or _headlesspi.py_ they both use _squishboxconf.yaml_, which is in the _SquishBox/_ directory.

**Reply by j-vanwijk** [reply]
Oh thanx that's explaining a lot. I have trying tot get it work it will only work on my headphones. So that's a step.

I will try today investigate the usb audio dongle to get it work.

Any results I will post here.

**Reply by j-vanwijk** [reply]
I can get it to work.
I had some problemen with latency to get fluidsynth to work. My Drangonfly usb stick has some optimal Sample rates so the Raspberry pi CPU is offloaded.  With "fluidsynth -a alsa -o audio.alsa.device=plughw:1,0 -m alsa_raw -o "midi.alsa.device=hw:0,0,0" sf2/piano/Kawai\ Upright\ Piano.sf2" I have get Fluidsynth to work nicely

for now it's for me fine. I've i got some spare time i can look howto get it work with the fluidpatcher and Alsa.
